pkg/proxy/router/taskrunner.go

```diff
@@ -83,11 +88,13 @@ func (tr *taskRunner) cleanupOutgoingTasks(err error) {
 	}
 }
 
+const pipelineBufSize = 512 * 1024
```

Comment:
why do we choose a 512kb buffer? what is the impact of a smaller or larger buffer?